File: subscriber.js
var zmq = require('zmq');

// Create a subscriber socket
// http://api.zeromq.org/2-1:zmq-socket#toc10
var sock = zmq.socket('sub');
var port = 5556;

// This will connect to an existing publisher bound to the
// following address...
sock.connect('tcp://127.0.0.1:' + port);

// Filters out messages by topic, this can be set to a
// blank string to receive all messages...
sock.subscribe('seconds.ping');

console.log('Consumer subscribed to port %s', port);

sock.on('message', function(msg) {
	
	// Separate out the topic and the message
	// data...
	var str = msg.toString();
	var pivot = str.indexOf(' ');
	var topic = str.substr(0, pivot);
	var raw = str.substr(pivot + 1);
	var data = JSON.parse(raw);

	console.log('Message: %s', topic, data);
});

File: subscriber.py
import zmq
import json

port = 5556
context = zmq.Context()

# Define this socket as a subscriber
# http://api.zeromq.org/2-1:zmq-socket#toc10
socket = context.socket(zmq.SUB)

# Will connect to a publisher bound to the following
# address...
# http://zeromq.github.io/pyzmq/api/zmq.html#zmq.Socket.connect
# http://api.zeromq.org/2-1:zmq-connect
socket.connect("tcp://127.0.0.1:%s" % port)

# This subscriber is listening out for all messages
# the string can be changed to filter by particular
# topics...
# http://zeromq.github.io/pyzmq/api/zmq.html#zmq.Socket.setsockopt_string
# http://api.zeromq.org/2-1:zmq-setsockopt
socket.setsockopt_string(zmq.SUBSCRIBE, 'seconds.ping')

while True:
	data = socket.recv().decode('UTF-8')
	topic, raw = data.split(' ', 1)
	message = json.loads(raw)
	print("Message: %s %s" % (topic, message))